packages/components/src/components/assay/actions.ts

```diff
@@ -25,7 +25,27 @@ import { AssayDefinitionModel, AssayUploadTabs, QueryGridModel, SchemaQuery } fr
 
 import { AssayUploadResultModel, IAssayUploadOptions } from './models';
 
-export function importAssayRun(config: AssayDOM.IImportRunOptions): Promise<AssayUploadResultModel> {
+export function fetchAllAssays(type?: string): Promise<List<AssayDefinitionModel>> {
+    return new Promise((res, rej) => {
+        Assay.getAll({
+            parameters: {
+                type,
+            },
+            success: (rawModels: any[]) => {
+                const models = List<AssayDefinitionModel>().asMutable();
```

Comment:
Likely not from your changes, but do you mind removing the asMutable() and asImmutable() usages here while we are touching this file?